src/global-firestore-data/overridden-fulfillment-choices.ts

```diff
@@ -134,3 +124,39 @@ export const addArbitraryOptIn = (
       },
     })
   );
+
+/**
+ * Precondition: the requirement id lists are disjoint
+ */
+export const resolveConflicts = (
+  courseUniqueID: string | number,
+  naturallyFulfilledRequirementIds: string[],
+  arbitraryOptInRequirementIds: string[],
+  optOutRequirementIds: string[]
+): void =>
+  updateRequirementChoice(
```

Comment:
There is too much code in this single function call IMO

Reply:
How do you think I should refactor it?